bootstrap/kubeadm/internal/ignition/clc/clc_test.go

```diff
@@ -0,0 +1,417 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package clc_test tests clc package.
+package clc_test
+
+import (
+	"testing"
+
+	ignition "github.com/flatcar-linux/ignition/config/v2_3"
+	"github.com/flatcar-linux/ignition/config/v2_3/types"
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/utils/pointer"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	"sigs.k8s.io/cluster-api/bootstrap/kubeadm/internal/cloudinit"
+	"sigs.k8s.io/cluster-api/bootstrap/kubeadm/internal/ignition/clc"
+)
+
+const (
+	configWithWarning = `---
+storage:
+  files:
+  - path: /foo
+    contents:
+      inline: foo
+`
+
+	// Should generate an Ignition warning about the colon in the partition label.
+	configWithIgnitionWarning = `---
+storage:
+  disks:
+  - device: /dev/sda
+    partitions:
+    - label: foo:bar
+`
+)
+
+func TestRender(t *testing.T) {
+	t.Parallel()
+
+	preKubeadmCommands := []string{
+		"pre-command",
+		"another-pre-command",
+		// Test multi-line commands as well.
```

Comment:
What does this comment mean? :) (same below)

Reply:
That we properly render multi-line pre commands, which wasn't the case in the past. Perhaps the comment can be re-phrased somehow.

Reply:
Ah nope, missed the \n somehow`. It's fine as is